iam/snippets/snippets_test.go

```diff
@@ -0,0 +1,53 @@
+// Copyright 2018 Google Inc. All rights reserved.
+// Use of this source code is governed by the Apache 2.0
+// license that can be found in the LICENSE file.
+
+package snippets
+
+import (
+	"math/rand"
+	"os"
+	"strconv"
+	"testing"
+	"time"
+)
+
+func TestViewGrantableRoles(t *testing.T) {
+	projectID := os.Getenv("GOLANG_SAMPLES_PROJECT_ID")
+	resource := "//cloudresourcemanager.googleapis.com/projects/" + projectID
+	viewGrantableRoles(resource)
```

Comment:
Pass in a buffer then check the output includes what you expect. Example: https://github.com/GoogleCloudPlatform/golang-samples/blob/master/monitoring/snippets/channel_delete_test.go#L27

Reply:
This reduces the usability of the snippet for the customer.

Reply:
Passed in buffer. For testing, I just wrote tests against the response directly.